MvvmCross/iOS/iOS/Views/Presenters/MvxIosViewPresenter.cs

```diff
@@ -111,48 +111,43 @@ public virtual void Show(IMvxIosView view, MvxViewModelRequest request)
             MvxRootPresentationAttribute attribute,
             MvxViewModelRequest request)
         {
+            // check if viewController is trying to initialize a navigation stack
+            if (attribute.WrapInNavigationController)
+            {
+                var navigationController = CreateNavigationController(viewController);
+                MasterNavigationController = navigationController as MvxNavigationController;
```

Comment:
We're clearing the presenter's reference to the old NavigationController by re-assigning it. That should be enough here... It will be disposed by the OS when appropiate